src/special-operators.jl

```
@@ -219,16 +223,18 @@ end
 eltypeof(op::AbstractLinearOperator) = eltype(op)  # need this for promote_eltypeof
 
 """
-    BlockDiagonalOperator(M1, M2, ..., Mn)
+    BlockDiagonalOperator(M1, M2, ..., Mn; S = Vector{promote_type(eltype.(M1, M2, ..., Mn))})
 
 Creates a block-diagonal linear operator:
 
     [ M1           ]
     [    M2        ]
     [       ...    ]
     [           Mn ]
+
+Change `S` to use LinearOperators on GPU.
 """
-function BlockDiagonalOperator(ops...) where {S}
+function BlockDiagonalOperator(ops...; S = Vector{promote_type(eltype.(ops)...)})
```

This should also raise an error if not concrete then, right?